### PR TITLE
docs: full-repository quality review reports with adversarial verification

### PR DESCRIPTION
**Pull request by cuioss-oliver**
## Summary

Adds `doc/quality-report/` — the outcome of a full-repository quality review covering all production code, all test code, and all documentation. **No repository code or documentation was modified**; the only change is the new report directory.

Review questions covered:
- Correctness and edge-case coverage of the code
- Hidden traps / false sense of security
- Whether the claimed security posture (fail-closed everywhere) is fully implemented
- Whether tests are real content tests (not "just coverage"), with correct generator usage and both positive and negative directions
- Whether documentation (Javadoc and AsciiDoc) is sensibly structured and matches the current code

## Contents

| Report | Scope |
|---|---|
| `README.adoc` | Methodology, executive summary, finding rollup |
| `code-review-validation.adoc` | Validation module: security core + support code (V-1…V-17) |
| `code-review-client.adoc` | OAuth/OIDC client module (C-1…C-8) |
| `code-review-quarkus-benchmarking.adoc` | Quarkus extensions, benchmarking, build (Q-1…Q-13) |
| `test-quality-review.adoc` | Test substance, generators, positive/negative coverage (TV-/TC-) |
| `documentation-review.adoc` | AsciiDoc accuracy, structure, link integrity (D-1…D-15) |
| `javadoc-review.adoc` | Javadoc accuracy and standards compliance (J-1…J-12) |
| `adversarial-verification.adoc` | Independent clean-slate re-verification of every finding |

## Key results

- **No fail-open token-acceptance path found anywhere** — the validation pipeline and client are consistently fail-closed.
- 30 major findings, none critical. Highlights: `maxTokenAgeSeconds` bypassed by the access-token cache (V-2); ECDH-ES JWE non-functional end-to-end (V-1, empirically verified); JWKS loader cannot recover from startup failures (V-3/V-4); documented "fails at startup" not implemented in the Quarkus extension (Q-3/Q-4); class-level `@BearerAuth` merge broken under ArC (Q-1); refresh-rotation family false-positive reuse detection (C-1); three Keycloak ITs pass for the wrong reason (TC-1…TC-3); seven non-compiling Javadoc usage examples (J-1…J-7); quick-start example that does not work as written (D-1).
- **Adversarial verification**: a clean-slate independent pass re-checked all 82 findings end-to-end — 80 CONFIRMED, 2 PARTIALLY CONFIRMED (detail corrections), 0 REFUTED. Two behavioral claims were reproduced empirically against the module's compiled classes.

## Notes for reviewers

- Reports reference review revision `d6b927c` (current `main`).
- Finding IDs are stable and cross-referenced between reports for follow-up issue tracking.

---
_Generated by [Claude Code](https://claude.ai/code/session_013U3cV6dXsLQrX6U4GZoSh6)_

## Summary by CodeRabbit

- **Documentation**
  - Added a comprehensive quality-report collection covering client, validation, Quarkus, benchmarking, documentation, Javadoc, and test quality.
  - Documented 82 reviewed findings with verification results, severity assessments, supporting evidence, risks, and confirmed strengths.
  - Added assessments of security controls, configuration, cryptography, testing coverage, documentation accuracy, and build behavior.
  - Included consolidated methodology, executive findings, repository status, and links to detailed supporting reports.

### CODE REVIEW

**Comment by cuioss-oliver** [comment]
Re: Sourcery review — both suggestions addressed in 453a39a. `README.adoc` now has a "Using This Report" section covering (1) how the finding IDs (V-/C-/Q-/TV-/TC-/D-/J-) are meant to be referenced from follow-up issues and fixing PRs, and (2) the snapshot semantics: the reports are a point-in-time snapshot of `d6b927c`, fixes should reference finding IDs rather than edit the reports in place (to preserve the adversarial-verification verdicts), and the directory should be re-generated wholesale or retired once a substantial portion of findings is resolved.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: doc/quality-report/README.adoc [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+No repository code or documentation was modified by this review; the only addition is this `doc/quality-report/` directory.
+
+== Executive Summary
+
+*Headline: no fail-open token-acceptance path was found anywhere.* The validation pipeline and the OAuth client are consistently fail-closed: every error path in token validation throws a typed exception; every ambiguous condition in the client aborts the flow. The adversarial test tiers on both sides are genuine — they drive production entry points with hand-crafted attack inputs and assert exact rejection outcomes. The documentation set has zero broken links across 77 AsciiDoc files, and its specification layer matches the code to an unusual degree.
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply doc/quality-report/README.adoc]: Fixed in f5e4f0d. The headline is now scoped to cryptographic token validation (signature, algorithm enforcement, issuer/audience/expiry, JWKS, JWE) and explicitly names the `maxTokenAgeSeconds` cache-policy exception (V-2) in the same sentence, so the summary no longer contradicts the finding it introduces.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: doc/quality-report/README.adoc [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+* *Client-side*: the refresh-token rotation family desynchronizes on a legitimate re-login/step-up and then destroys the session as false "reuse" — fail-closed, but a self-inflicted DoS that also revokes a valid grant (xref:code-review-client.adoc#C-1[C-1]).
+
+=== Test-suite reality check
+
+The suites are overwhelmingly real — but three Keycloak integration tests pass for the wrong reason (revocation, rotation, logout: xref:test-quality-review.adoc#TC-1[TC-1]–TC-3), a unit test named "algorithm confusion" tests something else (xref:test-quality-review.adoc#TV-1[TV-1]), the empty-`aud`-array security fork has zero tests (xref:test-quality-review.adoc#TV-2[TV-2]), and two documented caller contracts (state single-use, family-eviction replay) have neither enforcing code nor tests (TC-5, TC-6). Generator usage is consistent and correct where variation matters; the fixed alg/kid "generators" are documented as intentional but give a false impression of variation (TV-7).
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply doc/quality-report/README.adoc]: Fixed in f5e4f0d. The summary now distinguishes the two gaps: state single-use (TC-5) has neither enforcing code nor tests, while for family eviction (TC-6) the LRU cap is implemented and its mechanics tested — only the post-eviction replay case is untested.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: doc/quality-report/documentation-review.adoc [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+|===
+|Severity |*major* (first-contact user experience; broken example)
+|Confidence |high — verification verdict: PARTIALLY CONFIRMED (example brokenness confirmed; "fails at startup" corrected to "fails at first use", see xref:adversarial-verification.adoc[verification report])
+|Location |`README.md:92-97`; `token-sheriff-validation-quarkus/README.adoc` (~108-138, "Basic Configuration" and "Multiple Issuers" YAML examples)
+|===
+
+The minimal example shows only `sheriff.token.issuers.keycloak.jwks.http.well-known-url=…`. But issuers default to *disabled*: `IssuerConfigResolver.java:169-174` (`.orElse(false); // Default to disabled — require explicit enabled=true`) and line 122 throws `IllegalStateException("No enabled issuer configurations found")`. The extension's own config reference gets it right ("Issuers must be explicitly enabled (default is disabled)", `META-INF/quarkus-config-doc.adoc:14-15`). The README examples as written do not work: the `IllegalStateException` surfaces at first use — the first `@BearerAuth` request or health probe — rather than at deployment, because no eager startup validation exists (see xref:code-review-quarkus-benchmarking.adoc#Q-3[Q-3]).
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply doc/quality-report/documentation-review.adoc]: Fixed in f5e4f0d. The D-1 heading now reads "fails at first use" and the README summary line says the example "errors at first use/first probe", matching the adversarial-verification correction (consistent with Q-3: no eager startup validation exists).

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by cuioss-oliver** [comment]
@coderabbitai review

---
_Generated by [Claude Code](https://claude.ai/code)_
